uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/WorkPattern/Calendar.tpl

```diff
@@ -100,19 +99,20 @@
       </table>
     </div>
   {/section}
-  {literal}
-    <script type="text/javascript">
-      CRM.$(function($) {
-        var form = new CRM.HRLeaveAndAbsencesApp.Form.WorkPattern({/literal}'{$delete_url}'{literal});
+  <script type="text/javascript">
+    {literal}
+      CRM.$(document).on('CRMWorkPatternFormScriptIsReady', function () {
```

Comment:
Is this `CRMWorkPatternFormScriptIsReady` generated by Civi automatically?

Reply:
No, this is a custom event, the app JS file is now async and needs to be waited before executing what's inside the `CRM.$(document).on`